### ISSUE TITLE
forge init fails: fatal: pathspec '.gitmodules' did not match any files

### ISSUE THREAD

**Issue by miohtama** [foundry-rs/foundry#4552]
### Component

Forge

### Have you ensured that all of these are up to date?

- [x] Foundry
- [X] Foundryup

### What version of Foundry are you on?

forge 0.2.0 (a1f41be 2023-03-14T00:15:36.156534Z)

### What command(s) is the bug in?

 forge init

### Operating System

macOS (Apple Silicon)

### Describe the bug

[Following Creating a New Project documentation](https://book.getfoundry.sh/projects/creating-a-new-project), when running `forge init forge` I get the following error:

```shell
forge init forge
```

```
Initializing /Users/moo/code/executor/ethdubai-2023-hackathon/forge...
Installing forge-std in "/Users/moo/code/executor/ethdubai-2023-hackathon/forge/lib/forge-std" (url: Some("https://github.com/foundry-rs/forge-std"), tag: None)
Error: 
fatal: pathspec '.gitmodules' did not match any files
```

- Existing git repository
- macOS
- zsh

**Comment by miohtama** [comment]
The following command works:

```shell
forge init --no-git forge
```

My suggestion for the fix

- Detect that you are already in a git repository
- More helpful error message

**Comment by FreezyEx** [comment]
> Tracked it down to:
> 
> https://github.com/foundry-rs/foundry/blob/1e0b628d47a7fc18f4fbe0b0b84581fe6a9ad3b3/cli/src/cmd/forge/install.rs#L178-L180
> 
> Repro:
> 
> ```shell
> # or git init and add a .gitmodules
> forge init a
> cd a
> 
> forge init b
> # error: no .gitmodules ... because the gitmodules were added to the root one
> ```
> 
> @mattsse what's the best way to fix this? Recursing up until we find the `.gitmodules` file?

I had the same issue. To fix I deleted the git files and then run: `git init`

**Comment by brokenseal** [comment]
I get the same error when installing forget-std in a subfolder of my project.

**Comment by OxfordStreet** [comment]
I have the same issue, I just committed all the changes before install new libs.
I tried install `openzeppelin-contracts` and `solmate`, but there is only an error when I install the former :
### Install openzeppelin-contracts
```
MBP14:test_erc20 Ox$ forge install OpenZeppelin/openzeppelin-contracts
Installing openzeppelin-contracts in "/Users/reporterha/Documents/test_erc20/lib/openzeppelin-contracts" (url: Some("https://github.com/OpenZeppelin/openzeppelin-contracts"), tag: None)
Error: 
**fatal: pathspec '.gitmodules' did not match any files**
```
### Install solmate
```
MBP14:test_erc20 Ox$ forge install transmissions11/solmate
Installing solmate in "/Users/reporterha/Documents/test_erc20/lib/solmate" (url: Some("https://github.com/transmissions11/solmate"), tag: None)
    Installed solmate

```

**Comment by mattsse** [comment]
likely some git edge case, but hard to say without a repro

**Comment by stalinMacias** [comment]
I was having this exact issue when trying to use Foundry on a project that was already initialized to use git

I did the test to initialize a new foundry project on both, **a total clean empty folder**, and **under a folder that has already been initialized with git**

The result, **in the clean empty folder**, the `forge init --force` command **works as a charm**, and **in the other project I was getting the .git error reported on this thread**, the solution to **works with Foundry under a project that has already been initialized with git it is to always add** the `--no-git` **option at the end of each** `forge install` command. i.e.
```
forge install OpenZeppelin/openzeppelin-contracts --no-git
```

**Comment by kaxxa123** [comment]
There seems to be a problem with the detection of an existing git repository when ``forge init`` is run in a sub-dir.
Apart for reproducing the same problem I also noticed that ``forge init`` still fired a git commit despite it shouldn't be doing this as per fix #252

**Comment by Dayo-Adewuyi** [comment]
why's this closed when there is no solution yet

**Comment by DaniPopes** [comment]
This has been fixed in #5201. If you still face this, open another issue @Dayo-Adewuyi
